### PR TITLE
[SW-1312] Warn that default value of convertUnknownCategoricalLevelsToNa will be changed to false on GridSearch & AutoML

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala [hunk 1]
```diff
@@ -43,6 +43,8 @@ class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
 
   // Override default values
   setDefault(nfolds, 5)
+  logWarning("Default value of 'convertUnknownCategoricalLevelsToNa' parameter on H2OAutoML will be changed to 'false' from 'true' in the next" +
```
Comment (mn-mikke) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala]: I would remove 'on H2OAutoML'  and also avoid string concatenation.

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGridSearch.scala [hunk 2]
```diff
@@ -50,6 +50,8 @@ class H2OGridSearch(override val uid: String) extends Estimator[H2OMOJOModel]
   with H2OAlgorithmCommons with DefaultParamsWritable with H2OGridSearchParams {
 
   // Override default values
+  logWarning("Default value of 'convertUnknownCategoricalLevelsToNa' parameter on H2OGridSearch will be changed to 'false' from 'true' in the next" +
```
Comment (mn-mikke) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGridSearch.scala]: ditto

**Comment by mn-mikke** [comment]
@jakubhava Shouldn't we also raise warning on the Python site?

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala]: Thanks!

**Comment by jakubhava** [comment]
The warning on Python side is printed automatically from Scala

**Comment by mn-mikke** [comment]
Ok, cool!

**Comment by jakubhava** [comment]
@mn-mikke please have a look

**Pull request merged**